editor/src/components/canvas/canvas-strategies/strategies/set-padding-strategy.tsx

```diff
@@ -174,19 +176,51 @@ function supportsPaddingControls(metadata: ElementInstanceMetadataMap, path: Ele
     return false
   }
 
-  const childrenNotPositionAbsoluteOrSticky = MetadataUtils.getChildren(metadata, path).filter(
+  const children = MetadataUtils.getChildren(metadata, path)
+  if (children.length === 0) {
+    return true
+  }
+
+  const childrenNotPositionedAbsoluteOrSticky = MetadataUtils.getChildren(metadata, path).filter(
     (child) =>
       child.specialSizeMeasurements.position !== 'absolute' &&
       child.specialSizeMeasurements.position !== 'sticky',
   )
 
-  if (childrenNotPositionAbsoluteOrSticky.length < 1) {
+  if (childrenNotPositionedAbsoluteOrSticky.length < 1) {
+    return false
+  }
+
+  if (!elementHasPaddingSetFromMetadata(metadata, path)) {
     return false
   }
 
   return true
 }
 
+function elementHasPaddingSetFromMetadata(
+  metadata: ElementInstanceMetadataMap,
+  path: ElementPath,
+): boolean {
+  const element = MetadataUtils.getJSXElementFromMetadata(metadata, path)
+  if (element == null) {
+    return false
+  }
+
+  const jsxAttributeExists = (propertyPath: PropertyPath): boolean =>
+    getJSXAttributeAtPath(element.props, propertyPath).attribute.type !== 'ATTRIBUTE_NOT_FOUND'
+
+  const paddingProps = (p: keyof CSSPadding) => PP.create(['style', p])
+
+  return (
+    jsxAttributeExists(PP.create(['style', 'padding'])) ||
+    jsxAttributeExists(paddingProps('paddingBottom')) ||
+    jsxAttributeExists(paddingProps('paddingTop')) ||
+    jsxAttributeExists(paddingProps('paddingLeft')) ||
+    jsxAttributeExists(paddingProps('paddingRight'))
+  )
+}
```

Comment:
there's element.specialSizeMeasurements.padding that should give you 4 values that come from computedStyle – meaning we could _show_ the padding controls even if they are set from CSS or wherever.

Reply:
I tried this while working the PR, and it seems like that padding from specialSizeMeasurements is 0 when there's no padding set in the CSS.

We can make this work by only showing padding controls if any of the padding values is >0, but this leaves us with the slim edge case where all paddings are set to 0, but the still controls aren't shown. I felt this to go against the original spirit of the PR (to make props manipulable visually), so I went for the more verbose but more accurate solution.

Reply:
I refactored the strategy along your suggestion, and I added the tradeoff in the PR description for posterity